server/src/test/java/org/apache/iotdb/db/newsync/transport/TransportServiceTest.java

```diff
@@ -0,0 +1,193 @@
+        new SchemaPipeData(new SetStorageGroupPlan(new PartialPath("root.vehicle")), serialNum++));
+    pipeDataList.add(
+        new SchemaPipeData(
+            new CreateTimeSeriesPlan(
+                new PartialPath("root.vehicle.d0.s0"),
+                new MeasurementSchema("s0", TSDataType.INT32, TSEncoding.RLE)),
+            serialNum++));
+    TsFilePipeData tsFilePipeData = new TsFilePipeData(tsfile.getPath(), serialNum++);
+    pipeDataList.add(tsFilePipeData);
+    Deletion deletion = new Deletion(new PartialPath("root.vehicle.**"), 0, 33, 38);
+    pipeDataList.add(new DeletionPipeData(deletion, serialNum++));
+
+    // 3. start server
+    TransportServerManager.getInstance().startService();
+
+    // 4. start client
+    Pipe pipe = new TsFilePipe(createdTime1, pipeName1, null, 0, false);
+    TransportClient client = new TransportClient(pipe, "127.0.0.1", 5555);
+    for (PipeData pipeData : pipeDataList) {
+      client.senderTransport(pipeData);
+    }
+
+    // 5. check file
+    Thread.sleep(1000);
+    client.close();
+    TransportServerManager.getInstance().stopService();
+    File[] targetFiles = fileDir.listFiles((dir1, name) -> name.equals(tsfile.getName()));
+    Assert.assertNotNull(targetFiles);
+    Assert.assertEquals(1, targetFiles.length);
+    compareFile(targetFiles[0], tsfile);
+    File[] resourceFiles = fileDir.listFiles((dir1, name) -> name.equals(resourceFile.getName()));
+    Assert.assertNotNull(resourceFiles);
+    Assert.assertEquals(1, resourceFiles.length);
+    compareFile(resourceFiles[0], resourceFile);
+    File[] modsFiles = fileDir.listFiles((dir1, name) -> name.equals(modsFile.getName()));
+    Assert.assertNotNull(modsFiles);
+    Assert.assertEquals(1, modsFiles.length);
+    compareFile(modsFiles[0], modsFile);
+
+    // 6. check pipedata
+    tsFilePipeData.setTsFilePath(fileDir.getAbsolutePath() + File.separator + tsfile.getName());
+    ExecutorService es1 = Executors.newSingleThreadExecutor();
+    List<PipeData> resPipeData = new ArrayList<>();
+    es1.execute(
+        () -> {
+          for (int i = 0; i < pipeDataList.size(); i++) {
+            try {
+              resPipeData.add(pipeDataQueue.take());
+              pipeDataQueue.commit();
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+            }
+          }
+        });
+    try {
+      Thread.sleep(500);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    }
+    es1.shutdownNow();
+    Assert.assertEquals(pipeDataList.size(), resPipeData.size());
+    for (int i = 0; i < resPipeData.size(); i++) {
+      Assert.assertEquals(pipeDataList.get(i), resPipeData.get(i));
+    }
+  }
+
+  private void compareFile(File firFile, File secFile) {
+    try {
+      BufferedInputStream fir = new BufferedInputStream(new FileInputStream(firFile));
+      BufferedInputStream sec = new BufferedInputStream(new FileInputStream(secFile));
+      // 比较文件的长度是否一样
```

Comment:
// To compare the length of the files.
BTW, Why don't compare the hash of the files.

Reply:
Yeah, I compare the length and hash of the files now.